umm-spec-lib/test/cmr/umm_spec/test/xml_to_umm_mappings/distributed_format_util.clj

```diff
@@ -0,0 +1,209 @@
+
+  (testing "Split the string by ' or '. The input is a vector"
+    (let [test-string ["hello" "xml or html" "goodby"]]
+      (is (= ["hello" "xml" "html" "goodby"]
+             (util/parse-distribution-formats-split-by-or test-string))))))
+
+(deftest split-comma-test
+  "Test spliting a string or vector of strings by ','."
+
+  (testing "Split the string by ','."
+    (let [test-string "hello, good,goodby"]
+      (is (= ["hello" "good" "goodby"]
+             (util/parse-distribution-formats-split-by-comma-input-string test-string)))))
+
+  (testing "do not split the string by ', '."
+    (let [test-string "hello good goodby"]
+      (is (= "hello good goodby"
+             (util/parse-distribution-formats-split-by-comma-input-string test-string)))))
+
+  (testing "Split the string by ','. The input is a vector"
+    (let [test-string ["hello" "xml, html" "goodby"]]
+      (is (= ["hello" "xml" "html" "goodby"]
+             (util/parse-distribution-formats-split-by-comma test-string))))))
+
+(deftest split-dash-test
+  "Test spliting a string or vector of strings by ' - '."
+
+  (testing "Split the string by ' - '."
+    (let [test-string "hello - good - goodby"]
+      (is (= ["hello" "good" "goodby"]
+             (util/parse-distribution-formats-split-by-dash-input-string test-string)))))
+
+  (testing "do not split the string by '-'."
+    (let [test-string "hello-good- goodby -ok"]
+      (is (= "hello-good- goodby -ok"
+             (util/parse-distribution-formats-split-by-dash-input-string test-string)))))
+
+  (testing "Split the string by ' - '. The input is a vector"
+    (let [test-string ["hello" "xml - html" "goodby"]]
+      (is (= ["hello" "xml" "html" "goodby"]
+             (util/parse-distribution-formats-split-by-dash test-string))))))
+
+(deftest split-semicolon-test
+  "Test spliting a string or vector of strings by ';'."
+
+  (testing "Split the string by ';'."
+    (let [test-string "hello;good ; goodby"]
+      (is (= ["hello" "good" "goodby"]
+             (util/parse-distribution-formats-split-by-semicolon-input-string test-string)))))
+
+  (testing "do not split the string by ';'."
+    (let [test-string "hello ;amp good ;gt hello;amp ;gtgoodby ok;gt"]
+      (is (= ["hello ;amp good ;gt hello;amp ;gtgoodby ok;gt"]
+             (util/parse-distribution-formats-split-by-semicolon-input-string test-string)))))
+
+  (testing "Split the string by ';'. The input is a vector"
+    (let [test-string ["hello" "xml;html" "goodby"]]
+      (is (= ["hello" "xml" "html" "goodby"]
+             (util/parse-distribution-formats-split-by-semicolon test-string))))))
+
+(deftest parse-distribution-formats-test
+  "Test parsing a string by different split characters of (, and) (, or) (/) (and)
+   (or) (,) (-) and (;).  The initial input is a string, the output is either a
+   string or a vector of strings."
+
+  (testing "Split the string by all types."
```

Comment:
these are perfect cases for using `[cmr.common.util :refer [are3]]`.

Reply:
I am replacing what I can, but the macro expansion doesn't work correctly on all of my tests. In some cases it changes my expected output - which defeats the purpose of having a test.